plugin/pkg/scheduler/schedulercache/cache.go

```diff
@@ -233,6 +233,44 @@ func (cache *schedulerCache) RemovePod(pod *api.Pod) error {
 	return nil
 }
 
+func (cache *schedulerCache) AddNode(node *api.Node) error {
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+
+	n, ok := cache.nodes[node.Name]
+	if !ok {
+		n = NewNodeInfo()
+		cache.nodes[node.Name] = n
+	}
+	return n.SetNode(node)
+}
+
+func (cache *schedulerCache) UpdateNode(oldNode, newNode *api.Node) error {
```

Comment:
Yes - there is basically no difference now.

>
Adding the fast zero-allocation update path later when needed.

Sorry - I don't understand. Can you please clarify.